server/src/user/model.rs

```diff
@@ -20,34 +20,55 @@ pub struct UserData {
     pub password: String,
 }
 
-impl User {
-    fn generate_password(clear_password: &str) -> String {
-        bcrypt::hash(clear_password).unwrap()
-    }
+#[derive(Debug)]
+pub enum UserError {
+    UserNotFound,
+    UsernameTaken,
+    DatabaseError,
+    GenericError,
+}
```

Comment:
Is meant to reflect all errors occurring in the user model?

Reply:
Yes

Reply:
To give some further information when something goes wrong in the Model and also be able to react to some errors that are not bad (UsernameTaken)